loadtest/user/userentity/actions.go

```diff
@@ -339,6 +339,36 @@ func (ue *UserEntity) GetChannelMembers(channelId string, page, perPage int) err
 	return ue.store.SetChannelMembers(channelId, channelMembers)
 }
 
+// GetChannelMembersForUser gets all the channel members for a user on a team.
+func (ue *UserEntity) GetChannelMembersForUser(userId, teamId string) error {
+	channelMembers, resp := ue.client.GetChannelMembersForUser(userId, teamId, "")
+	if resp.Error != nil {
+		return resp.Error
+	}
+	if channelMembers == nil {
+		return errors.New("channelMembers should not be nil")
+	}
+	chanIdMap := make(map[string]*model.ChannelMembers)
+	for _, member := range *channelMembers {
+		if chanIdMap[member.ChannelId] == nil {
+			chanMembers := model.ChannelMembers([]model.ChannelMember{member})
+			chanIdMap[member.ChannelId] = &chanMembers
+		} else {
+			chanMembers := model.ChannelMembers(append(*chanIdMap[member.ChannelId], member))
+			chanIdMap[member.ChannelId] = &chanMembers
+		}
+	}
+
+	for chanId, members := range chanIdMap {
+		err := ue.store.SetChannelMembers(chanId, members)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
I kind of feel like this logic belongs to the `Store` layer.  
I propose we get rid of the redundant `channelId` parameter in `Store.SetChannelMembers()` and integrate this code there so that it would work for both API calls by using the data (`ChannelId`, `TeamId`) found in the `model.ChannelMember` structs.

Reply:
Agree.